xapian-core/tests/api_queryparser.cc

```diff
@@ -1725,6 +1725,84 @@ DEFINE_TESTCASE(qp_range4, !backend) {
     return true;
 }
 
+static const test test_value_filesizerange1_queries[] = {
+    { "10K..20K", "0 * VALUE_RANGE 1 10240 20480" },
+    { "10..50M", "0 * VALUE_RANGE 1 10485760 52428800" },
+    { "10K..20", "Unknown range operation" },
+    { "10A..20A", "Unknown range operation" },
+    { "10..20H", "Unknown range operation" },
+    { NULL, NULL }
+};
+
+// Test FileSizeRangeProcessor
+DEFINE_TESTCASE(qp_filesizerange1, !backend) {
+    Xapian::QueryParser qp;
+    Xapian::FileSizeRangeProcessor rp_filesize(1);
+    qp.add_rangeprocessor(&rp_date);
+    for (const test *p = test_value_filesizerange1_queries; p->query; ++p) {
+	string expect, parsed;
+	if (p->expect)
+	    expect = p->expect;
+	else
+	    expect = "parse error";
+	try {
+	    Xapian::Query qobj = qp.parse_query(p->query);
+	    parsed = qobj.get_description();
+	    expect = string("Query(") + expect + ')';
+	} catch (const Xapian::QueryParserError &e) {
+	    parsed = e.get_msg();
+	} catch (const Xapian::Error &e) {
+	    parsed = e.get_description();
+	} catch (...) {
+	    parsed = "Unknown exception!";
+	}
+	tout << "Query: " << p->query << '\n';
+	TEST_STRINGS_EQUAL(parsed, expect);
+    }
+    return true;
+}
+
+static const test test_value_filesizerange2_queries[] = {
+    { "size:10..20K", "0 * VALUE_RANGE 1 10240 20480" },
+    { "size:10..50M", "0 * VALUE_RANGE 1 10485760 52428800" },
+    { "filesize:10B..1M", "0 * VALUE_RANGE 2 10 1048576" },
```

Comment:
The expected results here are wrong - it should be the result of `Xapian::sortable_serialise()` on the number, like  `NumberRangeProcessor` would give - e.g.:

```
{ "1..12", "VALUE_RANGE 2 \\xa0 \\xae" },
```

As I said already (on IRC I think) you can get the correct encoded values by using `NumberRangeProcessor` on the scaled end points (so for `size:10..20K` look at `number:10240..20480`).